### PR TITLE
tests: Fix infinite recursion in __getattr__

### PR DESCRIPTION
**Pull request by rmcgibbo**
This fixes https://github.com/regebro/pyroma/issues/65

### CODE REVIEW

File: pyroma/tests.py
Comment (CAM-Gerlach) [inline pyroma/tests.py:92]: I'd suggest documenting this with a comment, and adding an appropriate error message to the AttributeError stating the name of the attribute and explaining what went wrong, like the others all have.

Comment (CAM-Gerlach) [reply pyroma/tests.py:92]: Also, should this error on all private or dunder attributes, all dunder attributes or only getattr calls to itself?

**Review by CAM-Gerlach — COMMENTED** [review]
@regebro 's the maintainer so I'd defer to him on approval, but I have a couple minor comments. Thanks!

File: pyroma/tests.py
Comment (rmcgibbo) [reply pyroma/tests.py:92]: > Also, should this error on all private or dunder attributes, all dunder attributes or only getattr calls to itself?

I'm not sure. I think the only thinks that are strictly necessary are the names like `_data` that are used inside the implementation of `__getattr__` itself. But I'm not actually familar with this code base, I just observed that this was sufficient to get the tests to pass.

Comment (rmcgibbo) [reply pyroma/tests.py]: I pushed a new version addressing your comments.

**Comment by regebro** [comment]
Yeah, this looks good, can you add yourself to the contributors, and also add a note in the CHANGES.txt?

**Comment by rmcgibbo** [comment]
> Yeah, this looks good, can you add yourself to the contributors, and also add a note in the CHANGES.txt?

I added a line to HISTORY.txt.

File: pyroma/tests.py
Comment (CAM-Gerlach) [reply pyroma/tests.py]: LGTM now, thanks.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM and looks like @regebro 's comments were addressed as well, at least to the extent possible in this project (as there is no AUTHORS file and the changelog is named HISTORY.txt; using standardized meta-file names following common convention will help avoid that and ensure users, GitHub, hooks and other tooling can find them will help avoid that, which I'll propose in a separate issue/PR).

**Pull request merged**
